piff/polynomial_interp.py

```diff
@@ -0,0 +1,389 @@
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+                           
+        :returns:          A numpy array of the calculated p_x(x)*p_y(y) where 
+                           the p functions are polynomials.
+
+        """
+        # Take the u and v components (x and y in the tangent plane)
+        # as our interpolants
+        u = pos[0]
+        v = pos[1]
+        # Call the appropriate function to generate the polynomial model.
+        # By default this is numpy.polyval. Note that despite appearances
+        # this is not a method call - function is a normal python attribute
+        # that happens to be a function.
+        f = self.function(u, v, C)
+        return f
+
+    def _initialGuess(self, positions, parameter, parameter_index):
+        """Make an initial guess for a set of parameters
+        to use in the fit for your model. This is passed
+        to curve_fit as a starting point.
+
+        :param positions:  A list of positions ((u,v) in this case) of stars.
+        :param parameter:  A numpy array of the measured values of a parameter
+                           for each star
+        :param parameter_index: The integer index of the parameter being used
+
+        :returns:          A guess for the parameters. In this case a 2D matrix
+                           which is zero everywhere except for (0,0).  This should
+                           correspond to a flat function of the parameters with 
+                           value given by the mean.
+
+
+        """        
+        # We need a starting point for the fitter.
+        # Use a constant value over the whole field as 
+        # a reasonable guess.
+        n = self.orders[parameter_index]+1
+        C = numpy.zeros((n,n))
+        C[0,0] = parameter.mean()
+        return C
+
+
+
+    def solve(self, pos, vectors, logger=None):
+        """Solve for the interpolation coefficients given some data,
+        using the scipy.optimize.curve_fit routine, which uses Levenberg-Marquardt
+        to find the least-squares solution.
+
+        This currently assumes that our positions pos are just u and v.
+
+        :param pos:         A list of positions to use for the interpolation.
+        :param vectors:     A list of parameter vectors (numpy arrays) for each star.
+        :param logger:      A logger object for logging debug info. [default: None]
+        """
+        import scipy.optimize
+
+        # We will want to index things later, so useful
+        # to convert these to numpy arrays and transpose
+        # them to the order we need.
+        parameters = numpy.array(vectors).T
+        positions = numpy.array(pos).T
+       
+        # We should have the same number of parameters as number of polynomial 
+        # orders with which we were created here.
+        nparam = len(parameters)
+        assert nparam==self.nparam, """Must create Polynomial interpolator 
```

Comment:
I think assert is the wrong thing here.  Asserts should be things that are always true if the code doesn't have bugs.  Not a way to check for user errors.  I'd rather this be
```
if nparam != self.nparam:
    raise ValueError("Must create Polynomial interpolator with ...")
```